src/main/java/org/minions/devfund/bruno/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,239 @@
+     *
+     * @param actorName String to set in Movie object.
+     * @param movieName String.
+     */
+    private void addActor(final String actorName, final String movieName) {
+        Actor actor = getAnActor(actorName);
+        if (actor != null) {
+            generatesTheMovies(actor, movieName);
+        } else {
+            Actor newActor = new Actor();
+            newActor.setName(actorName);
+            generatesTheMovies(newActor, movieName);
+            actorList.add(newActor);
+        }
+    }
+
+    /**
+     * Adds Rating to a movie.
+     *
+     * @param name   String movie.
+     * @param rating Double rating.
+     */
+    public void addRating(final String name, final double rating) {
+        Movie movie = getAMovie(name);
+        if (movie != null) {
+            movie.setRating(rating);
+        }
+    }
+
+    /**
+     * Updates the movie rating.
+     *
+     * @param name      String movie.
+     * @param newRating Double new Rating.
+     */
+    public void updateRating(final String name, final double newRating) {
+        Objects.requireNonNull(movieList.stream().filter(movie -> movie.getName().equals(name)).findFirst()
+                .orElse(null)).setRating(newRating);
+    }
+
+    /**
+     * Calculates the average from the List movies.
+     *
+     * @param movies List.
+     * @return average rating.
+     */
+    private double calculateTheAverageMovies(final List<Movie> movies) {
+        return movies.stream().mapToDouble(Movie::getRating).sum() / movies.size();
+    }
+
+    /**
+     * Gets the best Actor.
+     *
+     * @return String name actor.
+     */
+    public String getBestActor() {
+        String bestActor = "";
+        double bestAverage = 0;
+        for (Actor actor : actorList) {
+            if (calculateTheAverageMovies(actor.getMovies()) > bestAverage) {
+                bestAverage = calculateTheAverageMovies(actor.getMovies());
+                bestActor = actor.getName();
+            }
+        }
+        return bestActor;
+    }
+
+    /**
+     * Gets the best movie.
+     *
+     * @return String movie name.
+     */
+    public String getBestMovie() {
+        String bestMovie = "";
+        double majorRating = 0;
+        for (Movie movie : movieList) {
+            if (movie.getRating() > majorRating) {
+                majorRating = movie.getRating();
+                bestMovie = movie.getName();
+            }
+        }
+        return bestMovie;
+    }
+
+    /**
+     * Gets the actor list.
+     *
+     * @return actor List.
+     */
+    public List<Actor> getActorList() {
+        return actorList;
+    }
+
+    /**
+     * Gets the movie list.
+     *
+     * @return movie List.
+     */
+    public List<Movie> getMovieList() {
+        return movieList;
+    }
+
+    /**
+     * Manages the Actors and Movies generates according the movies and rating data base file.
+     */
+    public void main() {
```

Comment:
Consider removing this logic from the pull request.
It is not necessary even for edx.